### ISSUE TITLE
Add Ability to Open/Maximize Programs with Hotkeys

### ISSUE THREAD

**Issue by TheDoomCider** [Valkirie/HandheldCompanion#536]
I love Handheld Companion but I find the hotkey feature a bit limiting. Would you be able to add the ability to open/maximize programs like steam, epic, playnite, etc using a hotkey?

**Comment by CasperH2O** [comment]
Hello @TheDoomCider , there is a hotkey to Maximize a window, you can find it in the main GUI, hotkeys. It will require you configure it or pin and use it through QuickTools.

**Comment by CasperH2O** [comment]
> Thanks but what I meant was, I'd love to see a hotkey that can open Steam (for example) and if it is minimized/running in the background rather than opening it again, show it on screen. Much like what ayaspace or Asus's asus armoury crate does with their built in launchers.

I don't fully understand, do you mean to have a hotkey and when pressed it will minimize the current active foreground process (usually a game), remember which one it was and then the user does whatever the user needs to do in Windows, then press that same hotkey again and it will have remembered which process to maximize again?

I quite like that, I could see myself use that.

**Comment by TheDoomCider** [comment]
Not quite. I just want the ability to open a launcher like Steam using a hotkey.

**Comment by electronicrice** [comment]
> > Thanks but what I meant was, I'd love to see a hotkey that can open Steam (for example) and if it is minimized/running in the background rather than opening it again, show it on screen. Much like what ayaspace or Asus's asus armoury crate does with their built in launchers.
> 
> I don't fully understand, do you mean to have a hotkey and when pressed it will minimize the current active foreground process (usually a game), remember which one it was and then the user does whatever the user needs to do in Windows, then press that same hotkey again and it will have remembered which process to maximize again?
> 
> I quite like that, I could see myself use that.

A combination of this functionality you just described, with the ability to map a custom hotkey to an application executable for opening.

For example: in Custom Hotkeys, there'd be a 3rd option "Define application exe path". You put the path to the application, and you can also assign a button trigger like normal. When you press the button trigger, (or in my case I'd just pin it to Quick Tools), the application is launched. From there on, Quick Tools, or the button trigger, could remember the specific process launched and minimize/maximize as necessary.

**Comment by CasperH2O** [comment]
@electronicrice if I'm interpreting what you want correctly, it sounds more like you want a QuickTools page with an app/game launcher. Something like HCP has?

![image](https://github.com/Valkirie/HandheldCompanion/assets/14330834/0f5f0abc-fd4b-4fe5-9c8a-602ad0988bf3)

In general I like the idea of being able to open an app with a hotkey, either starting it, if it's already running, make it the focus window/maximize it.

**Comment by TheDoomCider** [comment]
Ooh, I like it! That would work for me!

**Comment by CasperH2O** [comment]
Current HC has a Quick Start feature, upcoming release will feature hotkeys to open apps: 

![image](https://github.com/Valkirie/HandheldCompanion/assets/14330834/bb1e45f5-67dd-4da8-9860-a748cd6dc53e)

![image](https://github.com/Valkirie/HandheldCompanion/assets/14330834/45b123cc-5911-4543-9711-495e11e93fe1)

But not the minimize, maximize feature.

**Comment by TheDoomCider** [comment]
Oooh, ok that is really awesome! Thanks! Once that feature is released, I will close this.
